runtime/calamari/src/migrations/staking.rs

```diff
@@ -0,0 +1,163 @@
+        let current_block = frame_system::Pallet::<T>::block_number() + 1u32.into();
+        pallet_parachain_staking::Pallet::<T>::initialize_pallet(
+            current_block,
+            invulnerables,
+            crate::currency::inflation_config::<T>(),
+        );
+        // Setting total_selected will take effect at the beginning of the next round, so for the first 6 hours
+        // our invulnerables will be the only collators
+        let _ = pallet_parachain_staking::Pallet::<T>::set_total_selected(
+            <T as frame_system::Config>::Origin::root(),
+            63u32,
+        );
+
+        // TODO: Get correct weight from the extrinsics
+        // T::DbWeight::get().reads_writes(reads, writes)
+        T::BlockWeights::get().max_block // only migration, we can just use the whole block
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        use crate::sp_api_hidden_includes_construct_runtime::hidden_include::traits::Currency;
+
+        // 2. Ensure they each have 400k KMA
+        let invulnerables = manta_collator_selection::Pallet::<T>::invulnerables();
+        for invulnerable in invulnerables.clone() {
+            assert!(
+                <T as pallet_parachain_staking::Config>::Currency::free_balance(&invulnerable)
+                    > T::MinWhitelistCandidateStk::get()
+            );
+        }
+
+        Self::set_temp_storage(invulnerables, "invulnerables");
+
+        Ok(())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() -> Result<(), &'static str> {
+        // Invulnerables were migrated correctly
+        let invulnerables: Vec<T::AccountId> =
+            Self::get_temp_storage("invulnerables").expect("must exist");
+        for invuln in invulnerables {
+            assert!(
+                !manta_collator_selection::Pallet::<T>::candidates()
+                    .iter()
+                    .any(|x| x.who == invuln),
+                "invulnerables must no longer be candidates with collator_selection",
+            );
+            let bond = pallet_parachain_staking::Bond {
+                owner: invuln.clone(),
+                amount: T::MinWhitelistCandidateStk::get(),
+            };
+            assert!(
+                pallet_parachain_staking::Pallet::<T>::candidate_pool().contains(&bond),
+                "invulnerables must now be candidates",
+            );
+            assert!(
+                pallet_parachain_staking::Pallet::<T>::selected_candidates().contains(&invuln),
+                "invulnerables must be active collators for the first round",
+            );
+        }
+        // Other pallet parameters are set
+        // Inflation is set to 3%
+        assert_eq!(
+            pallet_parachain_staking::Pallet::<T>::inflation_config()
+                .annual
+                .ideal,
+            sp_arithmetic::Perbill::from_percent(3)
+        );
```

Comment:
why not sanity check the other configurations in `InflationInfo` while you're at it

Reply:
good point, added